test/typescript/runtime.test.ts

```diff
@@ -0,0 +1,266 @@
+        // minify,
+        extend,
+        // deploy,
+        // localize,
+        // build,
+
+        reflect,
+        linked,
+        // infer,
+        builtin,
+        Association,
+        Composition,
+        entity,
+        event,
+        type,
+        array,
+        struct,
+        service,
+        // impl: (/** @type {(this:Service, srv:Service, ...etc) => any} */ srv) => srv,
+        // protocols: lazy => require('./srv/protocols'),
+        // bindings: lazy => require('./srv/bindings'),
+        // factory: lazy => require('./srv/factory'),
+        // /** @type Service[] */ providers: []
+
+        services,
+
+        server,
+        serve,
+        connect,
+        // middlewares,
+        // odata,
+        // auth,
+
+        // emit,
+
+        Service,
+        EventContext,
+        Request,
+        Event,
+        User,
+
+        // Services, Protocols and Periphery
+        ApplicationService,
+        MessagingService,
+        DatabaseService,
+        RemoteService,
+
+        // Contexts and Transactions
+        context,
+        spawn,
+        tx,
+
+        // Helpers
+        utils,
+        // error,
+        // exec,
+        test,
+        log,
+        debug,
+        lazify,
+        lazified,
+        // clone,
+        exit,
+
+        ql,
+        entities,
+
+        run,
+        foreach,
+        stream,
+        read,
+        create,
+        insert,
+        update,
+        delete : _delete,
+        // disconnect,
+        transaction,
+
+      } = cds
+    }
+
+    cds.service.impl(srv => {
+      srv.on('READ', 'Books', () => [{ID:1}])
+    })
+
+    cds.env.requires.db = { kind: 'sqlite' }
+    // let { Books } = cds.entities
+
+    expect (Service && cds.Service).toBe(Service)
+    expect (Request && cds.Request).toBe(Request)
+    expect (Event && cds.Event).toBe(Event)
+    expect (User && cds.User).toBe(User)
+    expect (cds.linked)
+
+    class MyService extends cds.ApplicationService {}
+    console.log (cds.linked(cds.parse(`entity Foo {}`)))
+    console.log (cds.linked.toString())
+
+    SELECT.from('Books').where({ID:1})
+
+    let ua = new User
+    expect (ua.id).toBe('anonymous')
+    expect(ua.is('any'))
+    expect(!ua.is('authenticated-user'))
+
+    let u2 = new cds.User('me')
+    expect (u2.id).toBe('me')
+    expect(u2.is('any'))
+    expect(u2.is('authenticated-user'))
+
+    let r = new Request ({event:'foo',data:{foo:'bar'},headers:{x:11}})
+    expect(r.event).toBe('foo')
+    expect(r.data.foo).toBe('bar')
+    expect(r.headers.x).toBe(11)
+
+    expect (cds.Association)
+    expect (cds.Composition)
+
+    cds.env.fiori.lean_draft = false
+    let Books = new cds.entity({name:'Books', elements: { HasDraftEntity:true }})
+    expect(Books.name).toBe('Books')
+    expect(Books.drafts!.name).toBe('Books_drafts')
+    let q = SELECT.from(Books).where({ID:1})
+    // expect(q.SELECT.from.ref[0]).toBe('Books') // FIXME: from.ref should work
```

Comment:
This is still a gap.  `SELECT.from.ref` is missing.
To be fixed

Reply:
[This PR](https://github.com/cap-js/cds-types/pull/10/files) seems to be related. But it apparently only contains a change to the test files? @johannes-vogel is there maybe a change to the corresponding .d.ts file you have not committed yet?

Reply:
I don't know of any.
→ similar issue is reproduced here: #10

Reply:
ah, so the PR is only meant to provide a test case, not a fix so far?

Reply:
yes, it was reported internally and I created the PR as a base for a fix.

Reply:
I see. Let's enrich your PR with a fix asap then. Maybe we can discuss this later in the types meeting.

Reply:
I have stacked [a PR](https://github.com/cap-js/cds-types/pull/22) on top of yours as a proposed solution. We can move on with this issue over there.